### DISCUSSION TITLE
GHome Smart Plug support

### DISCUSSION THREAD

**Discussion by rodajc** [arendst/Tasmota#20182] [Support request] [answered]
They claim that [this smart plug](https://www.amazon.de/GHome-Smart-Unterst%C3%BCtzung-Stromverbrauch-T%C3%9CV-Zertifiziert/dp/B09Q5BLMVT/) is supported by Tasmota, but... is it?

**Comment by barbudor** [comment]
If no one has tested before or taken care to update the template repository, who knows ?

Apparently as they make a difference between a normal pack and a Tasmota pack, I would assume they deliver them pre-flashed with Tasmota

That's Amazon, so free to try and send back if not working as expected
If it works, please update the template repository

**Reply by barbudor** [reply]
I'm interrested to know as they are cheaper than the Nous A1T

**Reply by sfromis** [reply]
They are only stating "integrated ESP8285 chip that supports firmware flashing" without any words to specifically suggest pre-flashed Tasmota. Looks like they are highlighting "ESP-Chip" as a feature, having noticed that some buyers wants this build. Of course, assuming that words can be taken quite literally may not be entirely safe.....

(Price is lower on Amazon.es)

**Reply by joba-1** [reply]
also my preferred form factor. 

From the description I would assume flashing is possible without too much hassle for an average amazon buyer, i.e. ota or preflashed.

Only drawback: only 10A while Nous A1T can do 16A

**Reply by sfromis** [reply]
The description does in no way suggest "average buyer", and neither OTA nor preflashed, **only** that it is flashable due to having ESP8285. I'd expect that they'd have stated it if it was preflashed, as an extra feature.

**Reply by barbudor** [reply]
![image](https://github.com/arendst/Tasmota/assets/2996042/714e2e7d-080b-46e7-b22e-fadf4fc12af6)

I would understand the Tasmota flavor to be a little more expensive to be pre-flashed

But there's only 1 way to know...

**Reply by sfromis** [reply]
They stated nothing such in the text. Wearing my cynical glasses, they may have realized that just having the old ESP8285 is a feature some buyers would pay extra for. And the prices are lower on Amazon.es. But yeah, only one way to know...

**Comment by rodajc** [comment]
> I'm interrested to know as they are cheaper than the Nous A1T

That is exactly the choice I have.

**Comment by rodajc** [comment]
> 
> From the description I would assume flashing is possible without too much hassle for an average amazon buyer, i.e. ota or preflashed.
> 

Yesterday I didn't even know Tasmota, so I am a pre-newby at best. I guess my first question is:
How would you flash a new firmware? Would it create an AP and present a GUI at port 80?

> Only drawback: only 10A while Nous A1T can do 16A

I noticed that too. 2300 W is still a lot.

**Reply by sfromis** [reply]
If it is not preflashed with Tasmota (you can still hope, I suppose), the procedure is to open it up, connect an USB-serial adapter to pads on the PCB, and run a tool to flash it.
https://tasmota.github.io/docs/Getting-Started/

**Comment by Jason2866** [comment]
It is clearly no Tasmota flashed. Tuya App driven. Needs disassembly and wired flashing. Until someone buys one, it is not known if it is the glued version or the nice one with the screw.

**Reply by rodajc** [reply]
If that is true you can hardly call that **Tasmota Flashing Unterstützung**

**Reply by sfromis** [reply]
They *can* as it will **support** Tasmota when having ESP8285, also without it being preflashed. Lots of people have done wired flashing to get commercial devices upgraded to Tasmota. When it is *possible*, the device can "support Tasmota". Question may be how easy this one is to open up. Note that even if it does have the screws for relatively easy disassembly, soldering to small pad is not what everyone thinks of as fun.

**Comment by rodajc** [comment]
> If it is not preflashed with Tasmota (you can still hope, I suppose), the procedure is to open it up, connect an USB-serial adapter to pads on the PCB, and run a tool to flash it.

But if that doesn't work you can't return it anymore so you don't want to do that, do you?

**Reply by barbudor** [reply]
Yes you can

**Reply by rodajc** [reply]
Return if after opening it up?

**Comment by rodajc** [comment]
I can only find ESP8266 firmware. But this smart plug has the ESP8285 chip, right?

**Reply by sfromis** [reply]
ESP8285 **is** an ESP8266, just with integrated flash on the chip, instead of needing two chips.

**Reply by rodajc** [reply]
Thanks!

**Comment by rodajc** [comment]
If flashing is impossible in an elegant manner, they wouldn't mention the chip nor Tasmota, because they don't want to have smart plugs returned to them that have been tortured and possibly even bricked. Does that make sense?

**Reply by sfromis** [reply]
It is not that unreasonable to expect that people actively wanting to run Tasmota are up for doing some tinkering, very much not "an average Amazon buyer", as lots of devices have been sold to users just wanting a device where it is *possible*, using wired flashing, which is a "normal thing to do" around Tasmota. My answer would be different if the intended audience was not techies.

However, I'd have liked if they were clear about preflashed or not, and maybe link to Tasmota for information about flashing, if not preflashed (my guess is no).

**Comment by Jason2866** [comment]
Maybe Tuya convert is possible. Manufacturers have the possibility to allow device specific downgrades of Tuya Firmware.

**Reply by rodajc** [reply]
My guess is that Tasmota is not preflashed but that the default smart plug firmware does have an OTA flash mode. Is that possible?

**Reply by sfromis** [reply]
That would not be my guess. If Jason is right about Tuya-Convert being *possible*, that would not be "an OTA flash mode", but a vulnerability to being "hacked" by that tool, exploiting security flaws fixed in newer versions of Tuya firmware. OTA, but not "elegant". Design allowing users to convert devices to Tasmota, without Tasmota being preflashed, would be a surprising thing.

**Reply by barbudor** [reply]
Everything is possible
Ask the vendor or order one to try
All other discussions are pointless and just guessing

**Comment by barbudor** [comment]
Looking at the customer's comment with the word "tasmota" shows that all comments from htis year reports the device is no more flashable with Tasmota as the ESP has been replaced
Only successfull comments are from 2022

**Reply by rodajc** [reply]
Amazon mixes comments on different items. Some complain about the USB outputs. This device doesn't have any USB.

**Reply by rodajc** [reply]
Ralf W. wrote in a review on Amazon only two days ago:

> Es ist ein ESP8285 Chip verbaut und die Dosen lassen sich ohne Mühe auf Tasmota flashen (OTA per Tuya-Convert).
> Sie werden nun direkt mit der Firmware 1.0.6 ausgeliefert und eine Installation einer Tuya-App ist somit zum Glück nicht mehr notwendig.

My German is a bit rusty but if firmware 1.0.6 is Tasmota (is it?), as far as I understand what he says is that it is running already without having to flash.

**Reply by barbudor** [reply]
There should be a button to translate in your own language 

> An ESP8285 chip is installed and the cans can be easily flashed on Tasmota (OTA via Tuya Convert).
 They are now delivered directly with firmware 1.0.6 and fortunately it is no longer necessary to install a Tuya app.

No 1.0.6 on Tasmota so unsure what he is talking about
May be 10.0.6?

**Reply by rodajc** [reply]
Hmm. Well, one of these days I will receive mine. Without any experience with Tasmota I should be able to find out what firmware it is shipped with and if it is OTA flashable to Tasmota.

**Comment by Jason2866** [comment]
The review you reference is mention the Tuya Firmware running on the device. If it is v1.0.6 it is the one with the security issue and can be converted to Tasmota with TuyaConvert (not a tool from Tasmota crew, no support).

**Comment by rodajc** [comment]
I received the plugs. The manual refers to the Gosund app. Installed the app and paired the first plug. In Device Update it says:
```
No updates available
Main Module: V1.0.6
MCU Module: V1.0.6

```
Next step Tuya convert? I'm on a laptop with archlinux. What is the safest procedure?

**Comment by Jason2866** [comment, marked as answer]
@rodajc TuyaConvert is here https://github.com/ct-Open-Source/tuya-convert
You get no support here for the tool.

**Comment by rodajc** [comment]
The first one (of 4) flashed successfully with TuyaConvert.

**Reply by barbudor** [reply]
👏 
Did you purchased the "Tasmota" ones or the cheaper ones ?

**Reply by rodajc** [reply]
I purchased the [EP2-A-4](https://www.amazon.de/GHome-Smart-Unterst%C3%BCtzung-Stromverbrauch-T%C3%9CV-Zertifiziert/dp/B09Q5BLMVT/).

**Reply by rodajc** [reply]
Any idea which config to try?

**Reply by barbudor** [reply]
I would try the templates for Nous A1T first: https://templates.blakadder.com/nous_A1T.html
{"NAME":"NOUS A1T","GPIO":[32,0,0,0,2720,2656,0,0,2624,320,224,0,0,0],"FLAG":0,"BASE":49}

If not working the old Nous A1 : https://templates.blakadder.com/nous_A1.html
{"NAME":"NOUS A1","GPIO":[320,0,576,0,2656,2720,0,0,2624,32,0,224,0,0],"FLAG":0,"BASE":45}

You can also follow that guide: https://tasmota.github.io/docs/Configuration-Procedure-for-New-Devices/

**Reply by rodajc** [reply]
It's working with the template for the **old Nous A1**.

**Reply by rodajc** [reply]
{"NAME":"GHome EP2-A","GPIO":[320,0,576,0,2656,2720,0,0,2624,32,0,224,0,0],"FLAG":0,"BASE":45}

**Reply by barbudor** [reply]
Do you get power measurment too ? Voltage Current and Power ?

**Reply by rodajc** [reply]
Yes. But all too high. Quick fix or [calibrate](https://tasmota.github.io/docs/Power-Monitoring-Calibration/#calibration-procedure)?

**Reply by rodajc** [reply]
Something like this?

`wget -qO - http://192.168.0.142/cm?cmnd=VoltageSet+230`

After setting the voltage both current and power are correct as well.

**Reply by barbudor** [reply]
Why using curl instead of the web console?
You live hard stuff?
🤣

**Reply by rodajc** [reply]
I know. The web console is cool too. But difficult to automate in a script.

Perhaps I should explain what I am going to use the smart plugs for.

They will be switching off electric heaters to prevent an overload. When the total power consumption, measured with [this power meter](https://github.com/rodajc/USB-power-meter), gets too high and an overload is imminent, because the electric oven is switched on e.g., one or more heaters are switched off to get the power consumption below the overload threshold.

**Reply by rodajc** [reply]
The script is going to run in an OpenWrt router. You don't want to depend on the cloud for this application...

**Reply by Noschvie** [reply]
Why not using an ESP32 running Tasmota to automate the control of the heaters?

**Reply by barbudor** [reply]
I understand the need for scripting your automation
But calibration is a one time operation

Your time, your choice

**Reply by rodajc** [reply]
Now I understand what you mean. Yes. You are right. The web console is perfect for the calibration procedure.

**Comment by rodajc** [comment]
Device template added. Problem solved. Thanks for your help everybody!

**Reply by tynor88** [reply]
Is it me, or is something missing in the template. It should be possible to change the color of the LED ring like it is on this device for instance: https://templates.blakadder.com/avatto_OT08.html

In the UI it has two options to control the intensity of the LED light and also the colour.

Thanks for all the great work, i also managed to flash them OTA with Tasmota!

**Reply by sfromis** [reply]
Assuming that you mean https://templates.blakadder.com/ghome_EP2-A.html the template is very different from when the Avatto one has gpios for 3 PWM channels, which Tasmota picks up as a RGB light. The GHome template is not very similar, different pins for energy monitoring, and having Led/LedLink pins which are status leds instead of a RGB light.

Unless the GHome plug actually has RGB channels internally, no lightshow for you. What I mean by that is that it would be possible for RGB leds to have two channels configured as status leds, using 2 of the 3 color channels. In that case, you'd have to explore the device to find the third color, and then modify the template for PWM pins.

**Comment by rodajc** [comment]
> Why not using an ESP32 running Tasmota to automate the control of the heaters?

How exactly? What I need is something like this:

```
if total power > limit then switch off heater1
if total power still > limit then also switch off heater2
etc...  

```
And they need to be switched back on again when the total power allows it.

The (bash) script is almost ready, as you see ;-) It must run somewhere on the LAN so the router is an excellent option.

**Reply by barbudor** [reply]
Even on ESP8266 you can use either
- Rules
- Scripting-Language

What is total power? Is that the power of this plug? Or the sum of multiple devices?

If your are using MQTT, it is easy for 1 tasmota device to receive energy message from other devices. With Scripting-Language you can do math and tests, then send power commands to other devices

**Reply by rodajc** [reply]
With total power I mean the power going into the [distribution board](https://en.wikipedia.org/wiki/Distribution_board) of the domestic installation. I measure it with a split-coil current transformer and an ATtiny45 microcontroller as explained [here](https://up.nl.eu.org/projects/sct.html) in a wider perspective. This power meter is connected to the router that can now control the heaters individually by switching on/off their Tasmota smart plug.

I didn't know Scripting-Language and MQTT is more powerful than I thought, but the web server running on the router provides a perfect way to monitor the bash script as well.

**Comment by rovo89** [comment]
I can confirm that tuya-convert works for the EP2-A-4. The only change I needed was adding `34987A` to `scripts/oui/esp.txt`.

@rodajc I switched "Led_i 1" to GPIO02 and "LedLinki" to GPIO00. To me, it makes more sense to show the power state with green LED and the WiFi status with the blue LED. What are the colors on your device?

**Reply by rodajc** [reply]
Because one of them is used in a bedroom I switched off the blue LED in the console. The green LED was off already.

You are right. What you propose makes more sense. Is that a change of the template?

**Reply by rovo89** [reply]
I also disable the LEDs when using the switch in the bedroom. But I'd still like to configure it "correctly" (subjective, I know) in case I want to use it elsewhere.

Here's the updated template (576 and 320 switched):
```
{"NAME":"GHome EP2-A","GPIO":[576,0,320,0,2656,2720,0,0,2624,32,0,224,0,0],"FLAG":0,"BASE":45}
```

**Reply by rovo89** [reply]
Shall I send a PR or do you want to do that?

**Reply by rodajc** [reply]
You have my **green light**, as I haven't got a clue how to do it. Thanks!

**Reply by rovo89** [reply]
Hehe. There's an "EDIT ON GITHUB" button on https://templates.blakadder.com/ghome_EP2-A.html. Since it's just a text file, it can be changed and committed in the browser. This is my pull request: https://github.com/blakadder/templates/pull/1854

**Reply by rodajc** [reply]
I get _You need to fork this repository to propose changes_ when I try to edit the file. And what is a pull request?

**Reply by rovo89** [reply]
I did some change before to that repository, so I had already forked it. Not sure if there's an option to fork it in the spot, that would make sense.

Pull request is the core of GitHub, it's the way to say "look, these are my changes, please include them". As I already did that, no more action required, you can just check the link above to see when it gets merged.

**Reply by rodajc** [reply]
Ok. I will. Thanks for your help and patience ;-)

**Reply by barbudor** [reply]
I put a comment on your PR because your proposed change is not functional but a personal preference
This alternate template should be provided as an alternate with explanations for future users to décide which to use.
Other wise we will have the same discussion in 6 month with someone who wants the opposite and end up in a PR that will yet suggest another configuration

**Reply by rodajc** [reply]
@rovo89 
I tried your modification to the template and the green LED indicates the power state now which is definitely more consistent with other devices and with the original firmware.

Now, the blue LED should indicate WiFi status, but does it though? Should the blue LED be on when it has an active WiFi connection? It is not here. Is it on your devices?

**Reply by rovo89** [reply]
I think it's only on when the connection failed: https://tasmota.github.io/docs/Lights/#link-status-led

**Reply by rodajc** [reply]
> Link status LED shows the network state, more specifically the Wi-Fi and MQTT connection status

I have a [Sonoff device](https://templates.blakadder.com/sonoff_Pow.html) with Tasmota and both LEDs are on.
Maybe the blue LED is on a different GPIO. Did you try?
